pkg/transform/oauth/oauth.go

```diff
@@ -227,3 +229,30 @@ func validateClientData(clientID string, clientSecret configv1.StringSource) err
 
 	return nil
 }
+
+func fetchSecret(secret configv1.StringSource, config *config.Config) (string, error) {
+	if secret.Value != "" {
+		return secret.Value, nil
+	}
+
+	if secret.File != "" {
+		secretFileContent, err := config.Fetch(secret.File)
+		if err != nil {
+			return "", nil
+		}
+
+		secretFileString := strings.TrimSuffix(string(secretFileContent), "\n")
+		return secretFileString, nil
+	}
+
+	if secret.Env != "" {
+		secretEnv, err := config.Env(secret.Env)
```

Comment:
`config.Env` is confusing because it's not related with tool configuration itself.